peloton/populate.py

```diff
@@ -0,0 +1,308 @@
+"""
+Populate https://github.com/orgs/SciTools/projects/13 from the results of query.py.
+"""
+
+from enum import Enum
+import json
+import random
+from string import ascii_lowercase
+from subprocess import run
+import sys
+from typing import List, NamedTuple
+
+import numpy as np
+import pandas as pd
+
+from common import csv_path, peloton_logins
+
+
+# https://docs.github.com/en/issues/planning-and-tracking-with-projects/
+#  automating-your-project/
+#  using-the-api-to-manage-projects#finding-the-node-id-of-a-user-project
+PROJECT_ID = "PVT_kwDOABU7f84ALhAI"
+
+# Number of commands to include in a single GraphQL query - avoids time-outs.
+PAGINATION = 20
+
+
+class MutationTypes(Enum):
+    # https://docs.github.com/en/graphql/reference/mutations
+    ADD = "addProjectV2ItemById"
+    DELETE = "deleteProjectV2Item"
+    UPDATE = "updateProjectV2ItemFieldValue"
+    CLEAR = "clearProjectV2ItemFieldValue"
+
+
+class DataTypes(Enum):
+    # https://docs.github.com/en/graphql/reference/
+    #  input-objects#projectv2fieldvalue
+    DATE = "date"
+    NUMBER = "number"
+    SELECT = "singleSelectOptionId"
+    TEXT = "text"
+
+
+class FieldInfo(NamedTuple):
+    """Key information on a project field to enable population."""
+    github_id: str
+    data_type: DataTypes
+    table_column: str
+
+
+class Fields(Enum):
+    """The project fields to populate."""
+    # https://docs.github.com/en/issues/planning-and-tracking-with-projects/
```

Comment:
I might be missing something, but this link can't be followed if it's over multilines?

Reply:
I felt this was preferable to a very long line. So long as it's obvious that it's a single link that you can input manually, I'm happy.

Reply:
Perhaps do it as a long comment, """? You can't just copy paste the link which is inconvenient

Reply:
Oooh that's cool, hadn't noticed you can copy-paste from multiple lines of plain text 👍

Reply:
5c0bdf6255e64edf361b5f26e3c2fc707f5ca5bc